dask_sql/physical/rel/logical/join.py

```diff
@@ -117,7 +116,7 @@ def merge_single_partitions(lhs_partition, rhs_partition):
                 # which is definitely not possible (java dependency, JVM start...)
                 lhs_partition = lhs_partition.assign(common=1)
                 rhs_partition = rhs_partition.assign(common=1)
-                merged_data = pd.merge(lhs_partition, rhs_partition, on=["common"])
+                merged_data = dd.multi.merge(lhs_partition, rhs_partition, on=["common"])
```

Comment:
```suggestion
                merged_data = lhs_partition.merge(rhs_partition, on=["common"])
```

This is a good idea, but it will not preduce a cudf DataFrame (which is what you would want for the output of a single partition), even if lhs_partition and rhs_partition are both cudf DataFrames.  This is because `dd.multi.merge` has an explicit check for `pd.DataFrame` instances, but not for cudf.

Reply:
That worked, and made the fix at https://github.com/dask/dask/pull/7990 unnecessary! (It also made it so we only have to call `.compute()` once, as desired.) Thanks!